app/util/decorators.py

```diff
@@ -27,6 +27,47 @@ def function_with_retries(*args, **kwargs):
     return method_decorator
 
 
+def retry_on_exception_exponential_backoff(exceptions, initial_delay=0.1, max_delay=10):
+    """
+    Binary exponential backoff.
+
+    :param exceptions: The exceptions that we will catch and retry on.
+    :type exceptions: list[Exception]
+    :param initial_delay: num seconds that the first retry period will be
+    :type initial_delay: float
+    :param max_delay: the max number of seconds that the retry period can go up
+    :type max_delay: float
+    """
+    def method_decorator(function):
+        @wraps(function)
+        def function_with_retries(*args, **kwargs):
+            # If initial_delay is negative, then exponentiation would go infinitely.
+            if initial_delay <= 0:
+                raise RuntimeError('initial_delay must be greater than 0, was set to {}'.format(str(initial_delay)))
+
+            # initial_delay must be smaller than max_delay.
+            if initial_delay > max_delay:
+                raise RuntimeError('initial_delay, {}, must be smaller than max_delay, {}'.format(
+                    str(initial_delay), str(max_delay)))
```

Comment:
When using `.format` you shouldn't need to cast to str -- that's one of the main features of it. :)